src/service/iot_config.proto

```diff
@@ -263,12 +276,17 @@ message route_stream_req_v1 {
 
 message route_stream_res_v1 {
   action_v1 action = 1;
-  route_v1 route = 2;
+  oneof data {
+    route_v1 route = 2;
```

Comment:
i think we'll need to still have the `update` action in the `action_v1` enum type so we can handle updates on the routes themselves; i think semantically we can get away with eui pairs and devaddr ranges only ever being added or removed but the routes seem like they need to support `update` as well

Reply:
I might be stuck on the way HPR is storing routes, but I'm trying to think of a situation where `update` would behave differently than `add`.

I only see `update` saying "you should already have this", but that's only guaranteed through agreement. Nothing should break if they get an update to something that doesn't exist.
But again, that's only thinking of HPR.

Reply:
Since the route type has field types that could change (maybe just server fields at this point I think) I could see updating the route to change the server port without making any other changes but if the consumer is going to treat that as a full delete/replace operation I can treat it as an update on the config service side and then broadcast it as an `add` operation as long as the HPR knows how to handle that record being attached to an existing route ID

Reply:
looking at the structure of a route message, i'm guessing that once a route is created the route_id, net_id, and the oui shouldn't change so we're really only talking about the max_copies field and the json structure of the server record that might update; if it's easier for the HPR to just receive `delete` requests to delete route records from ets and `add` requests to upsert records into ets and overwrite records if that route_id already exists that's easy enough for me